### PR TITLE
Upgraded API to Spring Boot 2.4.1.

### LINKED ISSUE #634: Upgrade API to Spring Boot 2.4.x
For various reasons but most pressingly because Okta's current SDK version integrates with Spring Security 5.4, not 5.3, and we do not want to be unable to get updates to the Okta SDK.

* Spring 2.4.0 release notes: https://spring.io/blog/2020/11/12/spring-boot-2-4-0-available-now
* Major issue we will have to deal with as part of 2.3->2.4 migration: https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-Config-Data-Migration-Guide
* Nice opportunity for streamlining: https://docs.spring.io/spring-boot/docs/2.4.0/gradle-plugin/reference/htmlsingle/#build-image-example-publish

### PR DESCRIPTION
**Pull request by benwarfield-usds**
Upgraded dependencies and added the necessary property setting to make sure we do not (yet) have to upgrade our properties files.

## Related Issue or Background Info

Resolves #634 
Rolls forward from incompatibilities that arose during #615

### CODE REVIEW

**Comment by benwarfield-usds** [comment]
Marking draft just to make double sure nobody accidentally YOLOs this.

**Review by timothybest-usds — APPROVED** [review]

File: backend/src/test/java/gov/cdc/usds/simplereport/service/BaseServiceTest.java [hunk 1]
```diff
@@ -107,7 +108,8 @@ protected void clearOktaUsers() {
     protected void clearOktaGroups() {
         for (Group g : _oktaClient.listGroups()) {
             String groupName = g.getProfile().getName();
-            if (groupName.startsWith(_authorizationProperties.getRolePrefix())) {
+            if (g.getType() == GroupType.OKTA_GROUP &&
+                        groupName.startsWith(_authorizationProperties.getRolePrefix())) {
```
Comment (jeremyzitomer-usds) [inline backend/src/test/java/gov/cdc/usds/simplereport/service/BaseServiceTest.java:112]: the same change should be made in BaseApiTest.java (though given the prognosis for Okta remaining in integration tests, maybe it doesn't matter too too much anyway)

Comment (benwarfield-usds) [reply backend/src/test/java/gov/cdc/usds/simplereport/service/BaseServiceTest.java:112]: Any changes you think are needed when we roll forward and want to add to the branch, I'm fine with—I just restored the things that you had to back out to make the older API version work. 😄

**Review by jeremyzitomer-usds — APPROVED** [review]
LGTM (even if sonar cloud doesn't agree 😜 )

**Pull request merged**